Tests/AWSLambdaRuntimeCoreTests/LambdaRuntimeClientTests.swift

```diff
@@ -86,4 +86,56 @@ struct LambdaRuntimeClientTests {
             }
         }
     }
+
+    @Test
+    func testCancellation() async throws {
+        struct HappyBehavior: LambdaServerBehavior {
+            let requestId = UUID().uuidString
+            let event = "hello"
+
+            func getInvocation() -> GetInvocationResult {
+                .success((self.requestId, self.event))
+            }
+
+            func processResponse(requestId: String, response: String?) -> Result<Void, ProcessResponseError> {
+                #expect(self.requestId == requestId)
+                #expect(self.event == response)
+                return .success(())
+            }
+
+            func processError(requestId: String, error: ErrorResponse) -> Result<Void, ProcessErrorError> {
+                Issue.record("should not report error")
+                return .failure(.internalServerError)
+            }
+
+            func processInitError(error: ErrorResponse) -> Result<Void, ProcessErrorError> {
+                Issue.record("should not report init error")
+                return .failure(.internalServerError)
+            }
+        }
+
+        try await withMockServer(behaviour: HappyBehavior()) { port in
+            try await LambdaRuntimeClient.withRuntimeClient(
+                configuration: .init(ip: "127.0.0.1", port: port),
+                eventLoop: NIOSingletons.posixEventLoopGroup.next(),
+                logger: self.logger
+            ) { runtimeClient in
+                try await withThrowingTaskGroup(of: Void.self) { group in
+                    group.addTask {
+                        while true {
+                            let (_, writer) = try await runtimeClient.nextInvocation()
+                            // Wrap this is a task so cancellation isn't propagated to the write calls
+                            try await Task {
+                                try await writer.write(ByteBuffer(string: "hello"))
+                                try await writer.finish()
+                            }.value
+                        }
+                    }
+                    // wait a small amount to ensure we are waiting for continuation
+                    try await Task.sleep(for: .milliseconds(100))
+                    group.cancelAll()
```

Comment:
we should check that the spawned task returns with a cancellation error indeed!

Reply:
It doesn't return cancellation. Currently returns isInClosedChannel. Not sure how to get it to return the cancellation error as I'm just closing the connection and waiting for it to clean up. I could catch the error and then check Task.isCancelled and throw a Cancellation error if it is cancelled but that seems a bit of a hack.